src/main/java/at/hannibal2/skyhanni/data/ActionBarStatsData.kt

```diff
@@ -1,29 +1,54 @@
 package at.hannibal2.skyhanni.data
 
+import at.hannibal2.skyhanni.events.ActionBarValueUpdate
 import at.hannibal2.skyhanni.events.LorenzActionBarEvent
 import at.hannibal2.skyhanni.utils.LorenzUtils
 import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import org.intellij.lang.annotations.Language
 
-object ActionBarStatsData {
-    // TODO USE SH-REPO
-    private val patterns = mapOf(
-        "health" to "§[c6](?<health>[\\d,]+)/[\\d,]+❤.*".toPattern(),
-        "defense" to ".*§a(?<defense>[\\d,]+)§a❈.*".toPattern(),
-        "mana" to ".*§b(?<mana>[\\d,]+)/[\\d,]+✎.*".toPattern(),
-        "riftTime" to "§[a7](?<riftTime>[\\dms ]+)ф.*".toPattern(),
+enum class ActionBarStatsData(@Language("RegExp") rawPattern: String) {
+    Health(
+        // language=RegExp
+        "§[c6](?<health>[\\d,]+)/[\\d,]+❤.*"
+    ),
+    defense(
+        // language=RegExp
+        ".*§a(?<defense>[\\d,]+)§a❈.*"
+    ),
+    mana(
+        // language=RegExp
+        ".*§b(?<mana>[\\d,]+)/[\\d,]+✎.*"
+    ),
+    riftTime(
+        // language=RegExp
+        "§[a7](?<riftTime>[\\dms ]+)ф.*"
+    ),
+    skyBlockXP(
+        // language=RegExp
+        ".*(§b\\+\\d+ SkyBlock XP §.\\([^()]+\\)§b \\(\\d+/\\d+\\)).*"
     )
+    ;
```

Comment:
can we get the enum names to be all capitals like the rest of the codebase and can you also add a trailing comma to the final one so we get cleaner changelogs in future if a new stat is added